src/main/java/graphql/schema/DataFetchingEnvironment.java

```diff
@@ -44,11 +45,9 @@
      * Returns a context argument that is set up when the {@link graphql.GraphQL#execute(String, Object)} method
      * is invoked
      *
-     * @param <T> you decide what type it is
-     *
      * @return a context object
      */
-    <T> T getContext();
+    ExecutionContext getContext();
```

Comment:
Now I am not sure we should actually expose ExecutionContext as API.

Reducing the API surface area is good for long term project health.

@mrcirillo - what do you want from ExecutionContext inside your data fetchers?

It currently has

         private final GraphQLSchema graphQLSchema;
    private final ExecutionId executionId;
    private final ExecutionStrategy queryStrategy;
    private final ExecutionStrategy mutationStrategy;
    private final Map<String, FragmentDefinition> fragmentsByName;
    private final OperationDefinition operationDefinition;
    private final Map<String, Object> variables;
    private final Object root;
    private final List<GraphQLError> errors = new CopyOnWriteArrayList<GraphQLError>();

which bits do you need?

I am not convinced that ExecutionContext is what DataFetchers need.  Convince me?

(We could invent a new API interface to contain what you need based on a convincing argument)